src/vtquery.cpp

```diff
@@ -219,6 +231,108 @@ std::vector<vtzero::property> get_properties_vector(vtzero::feature& f) {
     return v;
 }
 
+double convert_to_double(value_type value) {
+    // float
+    if (value.which() == 1) {
+        return double(boost::get<float>(value));
+    }
+    // double
+    if (value.which() == 2) {
+        return boost::get<double>(value);
+    }
+    // int64_t
+    if (value.which() == 3) {
+        return double(boost::get<int64_t>(value));
+    }
+    // uint64_t
+    if (value.which() == 4) {
+        return double(boost::get<uint64_t>(value));
+    }
+    return 0.0f;
+}
+
+/// apply filters to a feature
+bool filter_feature(vtzero::feature& feature, std::vector<filter_struct> filters) {
+    using key_type = std::string;
+    using map_type = std::map<key_type, value_type>;
+    auto map = vtzero::create_properties_map<map_type>(feature);
+    for (auto filter : filters) {
+        std::cout << "Search Key: " << filter.key << " ";
+        auto it = map.find(filter.key);
+        if (it != map.end())
+        {
+            std::cout << "Found Key: " << it->first << " ";
+            std::cout << "PT " << it->second.which() << " FT " << filter.value.which() << " ";
+            if (it->second.which() == 0 && filter.value.which() == 0) {
```

Comment:
String types support !=, ==, and contains, but doesn't do any string massaging like remove whitespace, casting to lowercase, or removing punctuation.